open-media-match/src/OpenMediaMatch/storage/mocked.py

```diff
@@ -98,8 +98,17 @@ def get_collab_data(
     def get_banks(self) -> t.Mapping[str, interface.BankConfig]:
         return dict(self.banks)
 
-    def bank_update(self, bank: interface.BankConfig, *, create: bool = False) -> None:
-        self.banks[bank.name] = bank
+    def bank_update(
+        self,
+        bank: interface.BankConfig,
+        *,
+        create: bool = False,
+        rename_from: t.Optional[str] = None,
+    ) -> None:
+        if create:
+            self.banks[bank.name] = bank
+        else:
+            self.banks[rename_from if rename_from is not None else bank.name] = bank
```

Comment:
blocking: this isn't quite correct - when rename from is true, you need to pop it out.

Try this:
```suggestion
        if rename_from is not None:
            self.banks.pop(rename_from)
        self.banks[bank.name] = bank
```

Reply:
yeah good catch :+1: